### ISSUE TITLE
whitespaces around operators

### ISSUE THREAD

**Issue by AiyionPrime** [mpeterv/luacheck#216]
I'm looking for an option to check for whitespaces around operators, but cannot find it, yet.
I did find a mention of this feature not being implemented yet, from 2016 though.

examples would be

- too less/many spaces around operator:

```lua
i=  3
```

- no spaces after `,`

```lua
for var_1,···,var_n in explist do body end
```
I realize its not an easy to implement check, due to its context sensitivity, but would like to state I'm missing this feature.
Maybe someone comes along feels the same, and leaves a :+1:.

This project is really nice to have around, thanks!

> [...]
> Ideally luacheck would also check for correct amount of whitespace around operators, parens, etc., but that would require more cooperation with the parser, and also it will be more complex to support customization. The warnings you suggested don't seem to need any options.
> 
> I haven't touched luacheck code in a while, and contributions are most welcome! I think that the way to do this would be to modify the lexer to consider whitespace a token, with exactly one whitespace token between each two consecutive normal tokens. Then the parser can call a function provided by the checker on each token. That function would check whitespace tokens and issue warnings.

_Originally posted by @mpeterv in https://github.com/mpeterv/luacheck/issues/79#issuecomment-260404880_

**Comment by ur4ltz** [comment]
See this!
https://github.com/mpeterv/luacheck/issues/198

**Comment by AiyionPrime** [comment]
Thanks for making me aware.

**Comment by AiyionPrime** [comment]
https://github.com/luarocks/luacheck/issues/19

**Comment by CppCXY** [comment]
Maybe [EmmyLuaCodeStyle](https://github.com/CppCXY/EmmyLuaCodeStyle) is good choice
